### ISSUE TITLE
Failed to install throuch pip or setup.py on OS X

### ISSUE THREAD

**Issue by jasonxxu** [dougalsutherland/cyflann#20]
Environment:
  * OS X El Capitan 10.11.6
  * Python: 2.7.11
  * Numpy: 1.11.1
  * Pip: 8.1.2
  * FLANN: stable 1.8.4 (installed via Homebrew)

Could not install by either:
```bash
pip install cyflann
```

Or setup.py after checking out the code:
```bash
python setup.py install
```

Error message was:
```
...
cyflann/index.c:254:10: fatal error: 'numpy/arrayobject.h' file not found
#include "numpy/arrayobject.h"
         ^
1 error generated.
error: command 'clang' failed with exit status 1
```

**Comment by dougalsutherland** [comment]
Thanks for the report / proposed fix!

- As discussed in the README, if you're using flann 1.8.4 you need to set `FLANN_DIR`; with homebrew it's `FLANN_DIR=$(brew --prefix)`. I added a workaround in69e970d3750f4715c5fa14574f2b09ab7b6ecf97 that should fix it for this case, though. It's also no longer an issue with the new flann 1.9.1; I'll send a PR to homebrew-science to update that.

- I can reproduce the numpy include error; I don't know why setting `include_dirs=[numpy.get_include()]` no longer works for that, but I changed it similarly to your proposed fix in 915a805fe48a16ae1a3b857c1eecadb207e82925.

**Comment by dougalsutherland** [comment]
FYI, PR to homebrew-science for the new FLANN version is https://github.com/Homebrew/homebrew-science/pull/4033.
